core/arch/arm/mm/sp_mem.c

```diff
@@ -46,6 +47,33 @@ struct sp_mem *sp_mem_new(void)
 	return smem;
 }
 
+bool sp_mem_is_shared(struct sp_mem_map_region *new_reg)
+{
+	struct sp_mem *smem = NULL;
+
+	SLIST_FOREACH(smem, &mem_shares, link) {
+		struct sp_mem_map_region *reg = NULL;
+
+		SLIST_FOREACH(reg, &smem->regions, link) {
+			uint64_t reg_end = reg->page_offset +
+					   (reg->page_count * SMALL_PAGE_SIZE);
+			uint64_t new_reg_end = 0;
+
+			new_reg_end = new_reg->page_offset +
+				      (new_reg->page_count * SMALL_PAGE_SIZE);
+
+			if (new_reg->mobj == reg->mobj) {
+				if (new_reg->page_offset < reg_end &&
+				    reg->page_offset < new_reg_end) {
```

Comment:
To me it would seem more natural to swap the order here like:
``` C
new_reg_end > reg->page_offset
```
You can keep it as it is if you think it's important.